internal/grader/reject.go

```diff
@@ -45,10 +45,48 @@ func (this *RejectWindowMax) fullString(now timestamp.Timestamp) string {
 		nextTime.SafeMessage(), deltaString)
 }
 
-func checkForRejection(assignment *model.Assignment, submissionPath string, user string, message string) (RejectReason, error) {
+type RejectLateWithoutAllow struct {
+	AssignmentName string
+	DueDate        timestamp.Timestamp
+}
+
+func (this *RejectLateWithoutAllow) String() string {
+	return this.fullString(timestamp.Now())
+}
+
+func (this *RejectLateWithoutAllow) fullString(now timestamp.Timestamp) string {
```

Comment:
RejectWindowMax.fullString() exists for tests.
If you aren't using this in tests, then only String() should exist.